pkg/repo/chartrepo.go

```diff
@@ -198,12 +200,13 @@ func (r *ChartRepository) generateIndex() error {
 // FindChartInRepoURL finds chart in chart repository pointed by repoURL
 // without adding repo to repositories
 func FindChartInRepoURL(repoURL, chartName, chartVersion, certFile, keyFile, caFile string, getters getter.Providers) (string, error) {
-	return FindChartInAuthRepoURL(repoURL, "", "", chartName, chartVersion, certFile, keyFile, caFile, getters)
+	return FindChartInRepoURLWithAuth(repoURL, "", "", chartName, chartVersion, certFile, keyFile, caFile, "", getters)
 }
 
 // FindChartInAuthRepoURL finds chart in chart repository pointed by repoURL
 // without adding repo to repositories, like FindChartInRepoURL,
 // but it also receives credentials for the chart repository.
+// Deprecated: this function is deprecated and will be removed in Helm 4, please use FindChartInRepoURLWithAuth instead.
```

Comment:
@mattfarina - do you think this is the best way to avoid modifying the public API? I'm not coming up with a good alternative given that this function takes other positional args such as "username" "password".

Should we take this moment to redesign the function to be more future proof? In the case there are more args? We can maybe use options to do things like `FindChartInRepoURLWithAuth(WithCaFile(...), WithToken(...) ...` instead?

Reply:
@jdolitsky I think this is a good way as you said. 
@mattfarina I can update my commit if you agree with this.